WIP: Updates to the press page

hola-soy-milk opened the pull request:
This aims to resolve #199 as discussed with @anasofiapinho and @alicetragedy.

hola-soy-milk committed:
Start drafting Logo section

hola-soy-milk committed:
Add link to press kit

hola-soy-milk committed:
Add images and image gallery draft

hola-soy-milk commented:
So here we have a first draft of image previews:
<img width="771" alt="screen shot 2016-08-13 at 23 57 26" src="https://cloud.githubusercontent.com/assets/656318/17645868/340d9604-61b2-11e6-8b11-4f43b94d6f46.png">

I added the image names into YAML data, so that we can dynamically add or remove images. Looks presentable on desktop, but on mobile...

<img width="361" alt="screen shot 2016-08-13 at 23 57 42" src="https://cloud.githubusercontent.com/assets/656318/17645875/4e4121e4-61b2-11e6-860a-0542566d8308.png">

Not so much. 😓 

On thing that occurs to me is that I'm using thumbnails versions of the images. A pain to generate but it's nice and loads the press page quickly. I could just crop the thumbnails to give them a consistent height. 

I won't do that yet, as I think it makes sense to first decide on what we'll do with images (Show them all? Select some?).

anasofiapinho commented:
Thanks @ramonh! <3
It's so good to start seeing these things coming together!

My feedback (risking that some of the topics might be redundant because you have just started):
- The images are stretched. How can we preview them with changing their dimensions in a non-proportional way?
- The page would have a more consistent feel to it if the width of the pictures was the same as the text.
- Wondering if we could put the button "Download the press kit" in the same level as the rest of the other buttons in the beginning of this section, next to images, videos, press releases...

hola-soy-milk commented:
Thanks for the feedback, @anasofiapinho!

We can definitely put one photo per "row". This would actually resolve the problem of the images being stretched, as they wouldn't need to have the same height.

Sorry, I didn't quite understand the third point, though. Do you mean at the top of the page?

anasofiapinho commented:
@ramonh sorry if I wasn't very clear, but I was asking if we could have 3 pictures per row and with the same width as the text.

I did a sketch. Not sure if I understood that it was this what we were going to do.

![img_2770](https://cloud.githubusercontent.com/assets/3285560/17670601/b1975106-630a-11e6-8e9a-52cc490b603c.JPG)

![img_2771](https://cloud.githubusercontent.com/assets/3285560/17671320/6906c1fc-630e-11e6-86eb-11ebb902e45d.JPG)

hola-soy-milk commented:
OH! Sorry, I didn't realize that I hadn't made it the same width as the text! Hahaha thank you for clarifying @anasofiapinho 😅 

Question, did we decide if we want to use all photos or only a handful? I ask because in order to make them look right, I'll be cropping them so they don't seem stretched 😄

hola-soy-milk commented:
Oh and thank you for the sketch too! This helps a lot 👍

anasofiapinho commented:
@ramonh I think we can pick some pictures, not all. If you are uploading them, I believe you should upload the picture with the award to show we have different types of pictures, not just team's pictures. Let me know if you need help!

anasofiapinho commented:
@ramonh What if on the last image square we just had a (grey) square with a text "More pictures" and when people clicked they would go to the images folder?

hola-soy-milk commented:
Ah good point, I'll put that one up too! 📷 

I like the second idea very much too! Should be totally doable

anasofiapinho commented:
Sorry @ramonh. I just noticed that I had uploaded the wrong image. I corrected my comment with the new image. Also, I started the drawing using "Pictures" but "Images" is better.

hola-soy-milk commented:
Ah that makes sense, thank you @anasofiapinho 😅

alicetragedy commented:
@ramonh using css, you should be able to force a width and height on the thumbnails but keep the width:height ratio intact by using a property like `overflow:hidden`. 
Regarding the yaml data, I think that's a great idea and is soooo much better than having html for each image! Do you think it would all be moved to a separate yaml file in `_data` so the html file doesn't get too busy?
As for the mobile view, I'd suggest having a row of two images max, if that's not too much trouble. 

For now I've only looked at the code and at your screenshots but didn't run it locally, I might have more comments then. Hope that helps!

hola-soy-milk commented:
Thanks @alicetragedy! I'll take another stab at this. I'm using `img` tags, which if I understood correctly don't really support `overflow:hidden`. 

I did, however, find this solution:
http://stackoverflow.com/a/18839446

It may not be very elegant to use inline styles, but this allows us to use the `yaml` data (which I'll definitely move into its own data file). What do you think?

I don't think it would be too much trouble to have two per row, but I think that'll involve media queries? I'll try it out 👍

alicetragedy commented:
@ramonh sounds great!!
Sorry, I realise now it might have been easier to comment directly on the lines of code 🙈 

As for the two per row: I'd say that's completely optional and shouldn't be a must — and yes, that would be with media-queries. But don't bother if you think it would be too much trouble! Just something that came to mind when looking at your screenshots 😅

hola-soy-milk commented:
No worries @alicetragedy! It's a discipline I've been putting off for far too long now.  🚂

hola-soy-milk committed:
Overhaul structure to prevent image stretching

This uses inline styling for backgrounds, as adapted from the following:
http://stackoverflow.com/a/18839446

hola-soy-milk committed:
Show 2 images per row on mobile

hola-soy-milk commented:
Update: Ahhh it works 🙈 

<img width="369" alt="screen shot 2016-08-15 at 20 31 02" src="https://cloud.githubusercontent.com/assets/656318/17674762/36c3b600-6327-11e6-85a1-18b6d29370ee.png">
<img width="766" alt="screen shot 2016-08-15 at 20 30 50" src="https://cloud.githubusercontent.com/assets/656318/17674763/36da72dc-6327-11e6-8e26-1775a0cce1df.png">

Looks pretty rad if you ask me. 
Thank you for the tips and feedback, @anasofiapinho and @alicetragedy! 🚀

hola-soy-milk committed:
Add missing whitespace

alicetragedy commented:
looks awesome!!!!! great work 👍 

![taylor-dancing-02](https://cloud.githubusercontent.com/assets/3143348/17674876/b21ad734-6327-11e6-924d-db0763a9f738.gif)

hola-soy-milk commented:
Thanks! I'll continue with the rest ☺️

anasofiapinho commented:
WOW! Go @ramonh! 🙌

anasofiapinho committed:
Add sections and materials

Sections and materials:
* Press Releases 3/4
* Videos
* Quotes
* Press Coverage links

I also made a new heading “Press Releases and Media” to separate the
first part “About” with the rest. I pushed the link to download the
press kit to this section and wrote an intro. See if that’s ok or if we
need to change it.

anasofiapinho commented:
New sections and materials:
- Press Releases 3/4
- Videos
- Quotes
- Press Coverage links

I also made a new heading “Press Releases and Media” to separate the
first part “About” with the rest. I pushed the link to download the
press kit to this section and wrote an intro. See if that’s ok or if we
need to change it.

anasofiapinho committed:
Add more articles to press coverage section

alicetragedy commented on about/press.html:
```diff
@@ -101,14 +101,67 @@
+      <br>
+      <h2>Press Releases and Media</h2>
+
+      <p>We gathered on the following sections a group of resources, such as press releases, logos, images, videos, quotes and some of the articles written about the program. Please feel free to use them, and in order to make the download easier, you can find the whole press kit <a href="https://www.dropbox.com/sh/ng9p8d5hba425ht/AAB18_F_NZzX_UlQL8LB1caMa?dl=0">here</a>.</p>
```
just fixing some wording, can you change to:
"In the following sections, we gathered a selection of resources.."
and 
"Please feel free to use them. In order to..."

alicetragedy commented on about/press.html:
```diff
@@ -101,14 +101,67 @@
+      <br>
       <h3>Images</h3>
 
       <p>Below are samples from teams of the Rails Girls Summer of Code during 2015 and 2016. There are many more in our Press Kit.</p>
```
change wording to:

"Below are samples from 2015 and 2016 Rails Girls Summer of Code teams.

hola-soy-milk committed:
Add links section with anchors

hola-soy-milk committed:
Rearrange and reword

alicetragedy commented on about/press.html:
```diff
@@ -102,12 +121,17 @@
 
       <br>
+      {% for link in page.links %}
+        <a href="#{{ link.href }}"><i class="fa fa-{{ link.icon }}"></i>{{ link.title }}</a>
```
super small change: could you add a space after the `</i>`? that way the icon and text aren't "squished" together. :D

hola-soy-milk replied:
Good catch, thanks @alicetragedy!

alicetragedy commented:
@ramonh just looked over your last changes, and made a comment (it's really nothing, just what I feel is a missing space between icon and link text). 
After that, I'm really really really ready to merge and would do so if Ana has no objections. Would love to ship this today!!

hola-soy-milk committed:
Add whitespace between icon and text in link

alicetragedy commented:
🎉

hola-soy-milk commented:
Thank you @alicetragedy! Two minor missing things are the styling of the links as well as the press release I'm wrapping up as we speak! I think this can be done today

alicetragedy commented:
Sorry, but I'm so eager to finally have this on our site 😂

alicetragedy commented on about/press.html:
```diff
@@ -58,7 +120,359 @@
+      <br>
+      <h3 id="images">Images</h3>
+
+      <p>Below are samples from teams of the Rails Girls Summer of Code during 2015 and 2016. There are many more in our Press Kit.</p>
```
As suggested by @anasofiapinho, since we have other types of images in dropbox, we might want to change the wording slightly to something like: 
Below are samples from teams of the Rails Girls Summer of Code during 2015 and 2016. There are many more, as well as other images, in our Press Kit.

anasofiapinho committed:
press-page-update

- Added Press releases links
- Added RGSoC logos and thumbnails
- Changed some photos and thumbnails because some were blurred
- Formatting font awesome icons
- General formatting
- Change some info in the about text
- Set up thumbnails and videos to open on a new window

anasofiapinho commented:
@alicetragedy and @ramonh, I just finished adding the rest of the materials and changing the text.
Let me know your opinion and please make all the changes you need.

Changes on this PR:
- Added Press releases links
- Added RGSoC logos and thumbnails
- Changed some photos and thumbnails because some were blurred
- Formatting font awesome icons
- General formatting
- Change some info in the about text
- Set up thumbnails and videos to open on a new window

alicetragedy commented:
@anasofiapinho @ramonh I think this is good to go — I'm ready to press the merge button!!!  :shipit:

hola-soy-milk commented:
Awesome! Thank you for your help

alicetragedy commented:
Aaaaaand it's a wrap! Great work @ramonh & @anasofiapinho 💪

hola-soy-milk commented:
Thank you for your help as well @alicetragedy

carpodaster commented:
Great work! Awesomesauce!
